### DISCUSSION TITLE
Why command are placed under the 'internal' folder

### DISCUSSION THREAD

**Discussion by chrjx** [photoprism/photoprism#1762] [General]
Sorry for inconvenience if the question is too simple. 

I just started to read the code of photoprism, I have a question for the project structure. Since the `cmd` folder is usually used for app commands, I am curious why command files such as `start.go`, `config.go` are placed under `internal` folder. 

Thanks in advance!

**Comment by lastzero** [comment]
see **https://eli.thegreenplace.net/2019/simple-go-project-layout-with-modules/**

**Reply by chrjx** [reply]
Thanks! It helps!

I misunderstood the usage of `cmd` folder. It is designed for executable program/command but excluding the command flags/options like `photoprism start`, `phototprism backup`. So the files such as `start.go`, `backup.go` should be placed elsewhere. Am I understanding correctly?

**Reply by lastzero** [reply]
Yes, larger apps usually have them somewhere in internal, for example in commands. Cmd is for the main package. You typically don't put everything in a main package unless it's a small program or proof of concept.
